analyzers/tests/SonarAnalyzer.UnitTest/Extensions/SyntaxNodeExtensionsTest.cs

```diff
@@ -722,6 +724,15 @@ public X M()
         public void Kind_Null_ReturnsNone() =>
             SonarAnalyzer.Helpers.SyntaxNodeExtensions.Kind<SyntaxKind>(null).Should().Be(SyntaxKind.None);
 
+        [DataTestMethod]
+        [DataRow("class Test { }", DisplayName = "When there is no pragma, return default file name.")]
+        [DataRow("#pragma checksum \"FileName.txt\" \"{guid}\" \"checksum bytes\"", "FileName.txt", DisplayName = "When pragma is present, return file name from pragma.")]
+        public void GetMappedFilePath(string code, string expectedFileName = DefaultFileName)
```

Comment:
I think we can test it with some actual generated files instead of hardcoding the directive ourselves, right?

Reply:
Yes, theoretically we can but the tests would be slower and the result would be the same. I would rely on IT's and peach for end-to-end validation.